src/common/function/FunctionManager.cpp

```diff
@@ -1388,43 +1388,41 @@ FunctionManager::FunctionManager() {
       if (args[0].get().isNull() || args[1].get().isNull() || args[2].get().isNull()) {
         return Value::kNullValue;
       }
+
       if (args[0].get().isStr() && args[1].get().isStr() && args[2].get().isStr()) {
-        std::string origStr(args[0].get().getStr());
-        std::string search(args[1].get().getStr());
-        std::string newStr(args[2].get().getStr());
-        return boost::replace_all_copy(origStr, search, newStr);
+        std::string &origStr = const_cast<std::string &>(args[0].get().getStr());
+        std::string search = args[1].get().getStr();
+        std::string newStr = args[2].get().getStr();
+        boost::replace_all(origStr, search, newStr);
+        return origStr;
       }
+
       if (args[0].get().isList()) {
-        auto list = args[0].get().getList();
+        auto &list = const_cast<List &>(args[0].get().getList());
```

Comment:
You might misunderstand my comment. `const_cast` is risky; we should not modify the value of the input parameter. Instead, we should copy it.

Reply:
Ok, I have removed the const_cast and changed it to directly copy the value of the input parameter for operation. At the same time, the list is modified in place directly instead of creating an additional result variable.